### PR TITLE
Fixed vcs_tag

### PR DESCRIPTION
**Pull request by ethauvin**
Also grab project description and url from pom, if not already specified.

See #419.

### CODE REVIEW

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt [hunk 1]
```diff
@@ -62,7 +62,7 @@ class BintrayApi @Inject constructor(val http: Http,
                            @Path("publish") publish: Int,
                            @Body file: File): Call<BintrayResponse>
 
-        class UpdateVersion(val desc: String?, val vcsTag: String?)
+        class UpdateVersion(val desc: String?, val vcs_tag: String?)
```
Comment (cbeust) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: I'd rather use the Gson annotation than renaming the Kotlin field (what is is, `@Field(name="")` or something like that?

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt [hunk 2]
```diff
@@ -96,25 +96,28 @@ class BintrayApi @Inject constructor(val http: Http,
...
                     .execute()
             if (result.errorBody() != null) {
                 throw KobaltException("Error while creating package:\n" + result.errorBody().string())
             }
         }
     }
 
-    private fun buildPackageInfo(project: Project): JsonObject {
+    private fun buildPackageInfo(project: Project, config: BintrayConfig): JsonObject {
         val jsonObject = JsonObject()
```
Comment (cbeust) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: You can cut down some boiler plate:


```
val jsonObject = JsonObject().apply {
  addNonNull(...)
}
```

**Comment by cbeust** [comment]
By the way, I don't see

> Also grab project description and url from pom, if not already specified.

in this PR.

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt [hunk 3]
```diff
@@ -112,9 +112,10 @@ class BintrayApi @Inject constructor(val http: Http,
     private fun buildPackageInfo(project: Project): JsonObject {
         val jsonObject = JsonObject()
         jsonObject.addNonNull("name", project.name)
-        jsonObject.addNonNull("desc", project.description)
+        jsonObject.addNonNull("desc",
+                if (project.description.isNotBlank()) project.description else project.pom?.description)
```
Comment (ethauvin) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: Pull the description from the pom if not specified.

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt [hunk 4]
```diff
@@ -112,9 +112,10 @@ class BintrayApi @Inject constructor(val http: Http,
...
         jsonObject.addNonNull("vcs_url", project.pom?.scm?.url)
-        jsonObject.addNonNull("website_url", project.url)
+        jsonObject.addNonNull("website_url", if (! project.url.isNullOrBlank()) project.url else project.pom?.url)
```
Comment (ethauvin) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: Pull the website url from the pom if not specified.

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt (hunk 1 above)
Comment (ethauvin) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: Does it really matter? The class is no exposed to the public.

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt (hunk 2 above)
Comment (ethauvin) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: Good idea.

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt (hunk 1 above)
Comment (cbeust) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: It hurts my eyes :-)

Comment (ethauvin) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: Don't look then. I'll fix it. ;-)

File: src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt [hunk 5]
```diff
@@ -62,7 +63,7 @@ class BintrayApi @Inject constructor(val http: Http,
                            @Path("publish") publish: Int,
                            @Body file: File): Call<BintrayResponse>
 
-        class UpdateVersion(val desc: String?, val vcs_tag: String?)
+        class UpdateVersion(val desc: String?, @SerializedName("vcs_tag") val vcsTag: String?)
```
Comment (ethauvin) [inline src/main/kotlin/com/beust/kobalt/plugin/publish/BintrayApi.kt]: `@Field("vcs_tag")` didng't work.

**Pull request merged**